### PR TITLE
logger: have separate filters for local and remote logging

### PR DESCRIPTION
**Pull request by bmwill**

### CODE REVIEW

File: common/logger/src/libra_logger.rs [hunk 1]
```diff
@@ -254,6 +295,7 @@ struct LoggerService {
     receiver: Receiver<LogEntry>,
     address: Option<String>,
     printer: Option<Box<dyn Writer>>,
+    facade: Arc<LibraLogger>,
```
Comment (gregnazario) [inline common/logger/src/libra_logger.rs:298]: Why not just call it `logger`?

File: common/debug-interface/src/node_debug_service.rs [hunk 2]
```diff
@@ -29,21 +29,43 @@ impl NodeDebugService {
...
+                    logger.set_remote_filter(Filter::builder().parse(filter).build());
                 }
 
                 warp::reply::reply()
             });
 
+        // Post /log
+        let log = warp::post()
+            .and(warp::path("log"))
+            .and(local_filter.or(remote_filter));
```
Comment (gregnazario) [inline common/debug-interface/src/node_debug_service.rs:67]: So this does both?

File: common/logger/src/libra_logger.rs [hunk 3]
```diff
@@ -132,6 +134,11 @@ impl LibraLoggerBuilder {
         self
     }
 
+    pub fn remote_level(&mut self, level: Level) -> &mut Self {
+        self.central_level = level;
+        self
```
Comment (gregnazario) [inline common/logger/src/libra_logger.rs:139]: Could we use the same terms for the `level` / `central_level` with the `local_level` / `remote_level`?

**Review by gregnazario — COMMENTED** [review]
It looks like everything should work correctly.  I'm just mostly concerned about settling on 2 names for everything here.  What I've got are there two types:
* `local`, `printer`, `level`
* `remote`, `writer`, `central_level`

I'd prefer we just say `local_level` and `remote_level` so it's consistent.  Or going with printer & writer everywhere.

**Review by gregnazario — APPROVED** [review]

File: common/debug-interface/src/node_debug_service.rs (hunk 2 above)
Comment (bmwill) [reply common/debug-interface/src/node_debug_service.rs:67]: `/log/filter` does the local and `/log/remote-filter` does the remote part

File: common/logger/src/libra_logger.rs (hunk 3 above)
Comment (bmwill) [reply common/logger/src/libra_logger.rs:139]: oops thanks for catching this

File: common/debug-interface/src/node_debug_service.rs (hunk 2 above)
Comment (gregnazario) [reply common/debug-interface/src/node_debug_service.rs:67]: Oh, I just wasn't sure what `filter.or(filter)` did both or what.

File: common/logger/src/libra_logger.rs (hunk 1 above)
Comment (bmwill) [reply common/logger/src/libra_logger.rs:298]: to be more clear since there's technically two different parts of the logger, the Service and the facade

**Pull request closed without merging**